src/Terminus.php

```diff
@@ -1,58 +1,81 @@
 <?php
 // src/Terminus.php
 
+namespace Pantheon;
+
+use Consolidation\AnnotatedCommand\AnnotatedCommandFactory;
+use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+use Symfony\Component\ClassLoader\Psr4ClassLoader;
 use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
 use Terminus\Config;
 
-class Terminus extends Application {
+class Terminus extends Application
+{
+  /**
+   * @var Config
+   */
+  private $config;
 
   /**
    * @inheritdoc
    */
-  public function __construct() {
-    $this->setAutoloader();
-    parent::__construct('Terminus', Config::get('version'));
-    date_default_timezone_set(Config::get('time_zone'));
-  }
+    public function __construct()
+    {
+        $this->config = new Config();
+        parent::__construct('Terminus', $this->config->get('version'));
+        date_default_timezone_set($this->config->get('time_zone'));
+    }
 
   /**
    * @inheritdoc
    */
-  protected function getDefaultHelperSet() {
-    $helper_set = parent::getDefaultHelperSet();
-    //$helper_set->set(new Terminus\Helpers\FileHelper());
-    return $helper_set;
-  }
+    public function run(InputInterface $input = null, OutputInterface $output = null)
+    {
+        $command_factory = new AnnotatedCommandFactory();
+        //$formatter_manager = new FormatterManager();
+
+        $discovery = new CommandFileDiscovery();
+        $command_files = $discovery->discover(__DIR__, '\Commands');
+        foreach ($command_files as $command_class) {
+            $command_instance = new $command_class(['config' => $this->config,]);
+            //$command_factory->commandProcessor()->setFormatterManager($formatter_manager);
+            $command_list = $command_factory->createCommandsFromClass($command_instance);
+            foreach ($command_list as $command) {
+                $this->add($command);
```

Comment:
I hate to jump in here because I know you just changed this, but I actually think it's more appropriate to inject these commands from outside the runner. The way you had it before. That's how most of the examples of Symfony console do it and it fits the inversion of control pattern of Symfony better.

I probably should have given this opinion earlier.. sorry 😬